### PR TITLE
Add diff syntax highlight

### PR DESCRIPTION
**Pull request by louim**
Hello!
Here is a screenshot what a diff looks like in the `Solarized Dark` theme
Screenshot 2023-04-20 at 2 07 15 PM
Here is the same diff in the `Spinel` theme
Screenshot 2023-04-20 at 2 07 19 PM

I added the required tokens to make it work with Spinel, [based on the Solarized dark colors.](https://github.com/microsoft/vscode/blob/233c88aede868ca6906e9b23842793f0196d32c5/extensions/theme-solarized-dark/themes/solarized-dark-color-theme.json#L224-L257)

Result:
image

I'd be happy to adjust the colors to something more appropriate for the theme, if someone want to suggest what should be used.

### CODE REVIEW

**Comment by louim** [comment]
I have signed the CLA!

**Comment by vinistock** [comment]
Thank you very much for the contribution! Are these tokens only used when viewing git patches or do they also apply to solving conflicts?

Also, instead of using colours from a different theme, we should try to stick to the colours used in Spinel. We have red, green and blue for keywords, strings and local variables, respectively. Can we use those instead?

File: themes/dark_spinel.json [hunk 1]
```diff
@@ -91,7 +91,6 @@
     {
       "name": "Punctuation, function and method calls",
       "scope": [
-        "punctuation",
```
Comment (vinistock) [inline themes/dark_spinel.json:94]: Let's not remove punctuation from here. Most of the time, we want punctuation to be the basic foreground colour. Instead, we can add more specific punctuation scopes to the colours you just added for the diffs.


The easiest way of discovering the scopes is by opening the command palette (CMD + SHIFT + P) and using `Inspect editor tokens and scopes`. If you try that in the patch, you'll discover the specific scopes you want to include are:
- `punctuation.definition.deleted.diff` (red)
- `punctuation.definition.inserted.diff` (green)
- `punctuation.definition.to-file.diff` (blue)

Not sure if there's any punctuation that would be associated with the orange. These three seem to do the trick

**Review by vinistock — CHANGES_REQUESTED** [review]

File: themes/dark_spinel.json (hunk 1 above)
Comment (louim) [reply themes/dark_spinel.json]: Good point, I wasn't sure what was the best strategy. I updated the commit with the required scopes.

**Review by vinistock — APPROVED** [review]
Awesome. Thanks again for contributing this! 🙏

**Pull request merged**